src/zenml/cli/base.py

```diff
@@ -64,18 +68,48 @@ def init(path: Optional[Path]) -> None:
 
 
 @cli.command("clean")
-@click.option("--yes", "-y", type=click.BOOL, default=False)
+@click.option("--yes", "-y", is_flag=True, default=False)
 def clean(yes: bool = False) -> None:
-    """Clean everything in repository.
+    """Delete all ZenML metadata and artifacts.
+
+    This is a destructive operation, primarily intended for use in development.
 
     Args:
       yes: bool:  (Default value = False)
     """
     if not yes:
-        _ = confirmation(
-            "This will completely delete all pipelines, their associated "
-            "artifacts and metadata ever created in this ZenML repository. "
+        confirm = confirmation(
+            "DANGER: This will completely delete all artifacts and metadata ever created \n"
+            "in this ZenML repository. Note: Pipelines and stack components running \n"
+            "non-locally will still exist. Please delete them manually.\n"
             "Are you sure you want to proceed?"
         )
 
-    error("Not implemented for this version")
+    if yes or confirm:
+        local_zen_repo_config = Path.cwd() / REPOSITORY_DIRECTORY_NAME
+        global_zen_config = Path(get_global_config_directory())
+        if fileio.exists(str(local_zen_repo_config)):
+            fileio.rmtree(str(local_zen_repo_config))
+            declare(f"Deleted local ZenML config from {local_zen_repo_config}.")
+        if fileio.exists(str(global_zen_config)):
+            config_yaml_path = global_zen_config / CONFIG_FILE_NAME
+            config_yaml_data = yaml_utils.read_yaml(str(config_yaml_path))
```

Comment:
you don't have to load the yaml file, you already have all the information in the GlobalConfiguration() singleton object, so you can just use something like `old_user_id = GlobalConfiguration().user_id` instead.

You should also preserve the `analytics_opt_in` value.